wagtail/admin/views/generic/models.py

```diff
@@ -715,6 +721,114 @@ def get_context_data(self, **kwargs):
         return context
 
 
+class InspectView(PermissionCheckedMixin, WagtailAdminTemplateMixin, TemplateView):
+    template_name = "wagtailadmin/generic/inspect.html"
+    page_title = gettext_lazy("Inspecting")
+    model = None
+    edit_url_name = None
+    delete_url_name = None
+    fields = []
+    fields_exclude = []
+    pk_url_kwarg = "pk"
+
+    def setup(self, request, *args, **kwargs):
+        super().setup(request, *args, **kwargs)
+        self.pk = self.kwargs[self.pk_url_kwarg]
+        self.fields = self.get_fields()
+        self.object = self.get_object()
+
+    def get_object(self, queryset=None):
+        return get_object_or_404(self.model, pk=unquote(self.pk))
+
+    def get_page_subtitle(self):
+        return str(self.object)
+
+    def get_fields(self):
+        fields = self.fields or [
+            f.name
+            for f in self.model._meta.get_fields()
+            if f.concrete
+            and (not f.is_relation or (not f.auto_created and f.related_model))
+        ]
+
+        fields = [f for f in fields if f not in self.fields_exclude]
+        return fields
+
+    def get_field_label(self, field_name, field):
+        return capfirst(label_for_field(field_name, model=self.model))
+
+    def get_field_display_value(self, field_name, field):
+        # First we check for a 'get_fieldname_display' property/method on
+        # the model, and return the value of that, if present.
+        value_func = getattr(self.object, "get_%s_display" % field_name, None)
+        if value_func is not None:
+            if callable(value_func):
+                return value_func()
+            return value_func
+
+        # Now let's get the attribute value from the instance itself and see if
+        # we can render something useful. Raises AttributeError appropriately.
+        value = getattr(self.object, field_name)
+
+        if isinstance(value, models.Manager):
+            value = value.all()
+
+        if isinstance(value, models.QuerySet):
+            return ", ".join(str(obj) for obj in value) or "-"
+
+        display_class = display_class_registry.get(field)
+
+        if display_class:
+            return display_class(value)
+
+        return value
+
+    def get_context_for_field(self, field_name):
+        try:
+            field = self.model._meta.get_field(field_name)
+        except FieldDoesNotExist:
+            field = None
+        context = {
+            "label": self.get_field_label(field_name, field),
+            "value": self.get_field_display_value(field_name, field),
+            "component": None,
+        }
+        if isinstance(context["value"], Component):
+            context["component"] = context["value"]
```

Comment:
Not very happy with this, but the value returned by `get_field_display_value` may not always be a `Component` class as it also handles arbitrary attributes on the model (not just model fields).

I suppose the nice thing about this is that it also lets you return a `Component` instance from a model method.

The main reason why I needed this was so that the template knows when to use `{% component %}`. If you have better ideas let me know!